src/app/tabs/run-tab.js

```diff
@@ -167,26 +164,45 @@ var css = csjs`
 
 module.exports = runTab
 
-var instanceContainer = yo`<div class="${css.instanceContainer}"></div>`
 var noInstancesText = yo`<div class="${css.noInstancesText}">0 contract Instances</div>`
 
-var pendingTxsText = yo`<div class="${css.pendingTxsText}"></div>`
-var pendingTxsContainer = yo`<div class="${css.pendingTxsContainer}">${pendingTxsText}</div>`
-
 function runTab (container, appAPI, appEvents, opts) {
+  var self = {
+    _view: {},
+    addInstance: addInstance
+  }
+  var udapp = appAPI.udapp()
+
+  var instanceContainer = yo`<div class="${css.instanceContainer}"></div>`
+
+  var pendingTxsText = yo`<div class="${css.pendingTxsText}"></div>`
+  var pendingTxsContainer = yo`
+    <div class="${css.pendingTxsContainer}">
+      <div class=${css.buttons}>${makeRecorder(self, appAPI, appEvents)}</div>
+      ${pendingTxsText}
+    </div>
+  `
+
   var el = yo`
   <div class="${css.runTabView}" id="runTabView">
-    ${settings(appAPI, appEvents)}
-    ${contractDropdown(appAPI, appEvents, instanceContainer)}
+    ${settings(self, appAPI, appEvents)}
+    ${contractDropdown(self, appAPI, appEvents, instanceContainer)}
     ${pendingTxsContainer}
     ${instanceContainer}
   </div>
   `
   container.appendChild(el)
 
+  function addInstance (sourcename) {
+    var contract = appAPI.getContract(sourcename)
+    var address = self._view.atAddressButtonInput.value
```

Comment:
that why you get `0x...0x` in the contract instance. here you should get the created contract address...